gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/web/auth/handler/impl/OAuth2AuthHandlerImpl.java

```diff
@@ -97,36 +107,52 @@ public void handle(RoutingContext context) {
                     context.put(ConstantKeys.CLIENT_CONTEXT_KEY, client);
                 }
 
-                // check if current subject can access its own resources
-                if (selfResource) {
-                    final String resourceId = context.request().getParam(resourceParameter);
-                    boolean isResourceIdValid = resourceId != null && resourceId.equals(token.getSub());
-                    boolean hasRequiredScope = resourceRequiredScope == null || token.hasScope(resourceRequiredScope);
+                var single = Single.just(Optional.of(token.getSub()));
+                if (selfResource && isSelfResourceAUser) {
+                    single = this.subjectManager.findUserIdBySub(token.getSub())
+                            .map(Optional::ofNullable)
+                            .switchIfEmpty(single);
+                }
 
```

Comment:
Maybe<Optional<?>> can be simplified
`                Maybe<String> single = Maybe.just(token.getSub());
                if (selfResource && isSelfResourceAUser) {
                    single = this.subjectManager.findUserIdBySub(token.getSub())
                            .switchIfEmpty(single);
                }`